src/ImageSharp.Web/Commands/PresetOnlyQueryCollectionRequestParser.cs

```diff
@@ -31,31 +31,33 @@ public class PresetOnlyQueryCollectionRequestParser : IRequestParser
             this.presets = ParsePresets(presetOptions.Value.Presets);
 
         /// <inheritdoc/>
-        public IDictionary<string, string> ParseRequestCommands(HttpContext context)
+        public CommandCollection ParseRequestCommands(HttpContext context)
         {
             if (context.Request.Query.Count == 0 || !context.Request.Query.ContainsKey(QueryKey))
             {
-                return new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                return new();
```

Comment:
Can you use a static instance here? Maybe better for no-ops

Reply:
I’ll have a look. Should be possible

Reply:
Would make sense to add it as `CommandCollection.Empty` 👍🏻

Reply:
Actually this is correct as it is. The parser returns an empty collection that can then be augmented via `OnParseCommandsAsync` we don't want a shared instance for that.